godel_surface_detection/src/detection/surface_detection.cpp

```diff
@@ -367,18 +253,23 @@ bool SurfaceDetection::find_surfaces()
 	std::vector<pcl::PointIndices> clusters_indices;
 	std::vector<Normals::Ptr> segment_normals;
 
+	// Pass through filter the data to constrain it to our ROI
+  pcl::PassThrough<pcl::PointXYZ> pass;
+  pass.setInputCloud(process_cloud_ptr);
+  pass.setFilterFieldName("z");
+  pass.setFilterLimits(-0.1, 0.4);
```

Comment:
Absolutely. The correct answer is to have them in the godel surface detection parameter msg.

That message needs something of an overhaul because there are a few invalid fields in it now. So I'll move this to a class level variable and we'll deal with the parameter in another structure.